### ISSUE TITLE
BUG: `tune=False` is not respected for sequential sampling

### ISSUE THREAD

**Issue by covertg** [pymc-devs/pymc#6512]
### Describe the issue:

Single process MCMC sets `step.tune = True` any time the number of tuning steps is nonzero (due to [this line](https://github.com/pymc-devs/pymc/blob/28bac770c138a9f369765478471df2a45c5ae596/pymc/sampling/mcmc.py#L812-L814)).

Discovered this when trying to debug a model that was throwing exceptions during tuning, regardless of whether or not I enabled or disabled tuning at the step method constructor.

This is not the same behavior as multiprocess MCMC (which does no such adjustment). And further, it has different effects depending on the step method. For DEMetropolis(Z), setting `tune=True` actually doesn't cause tuning to happen, because the tuning target is set to None (rather than lambda or scaling). But in Slice, and I think most others, tuning does get turned back on.

Due to these discrepancies I suppose this is a bug.

### Reproduceable code example:

```python
<error noticed in the debugger>
```

### PyMC version information:

```
%watermark -p pymc,pytensor
pymc    : 5.0.2
pytensor: 2.9.1
```

**Comment by ricardoV94** [comment]
I don't remember well, but isn't tuning disabled by setting tune=0 when you call `sample`?

**Comment by covertg** [comment]
Right, which just brings the (slight) inconvenience of having to discard tuning/"burn-in" samples after the fact. Maybe this is a less common use, but this would also prevent someone from selectively tuning some step methods but not others (if using compoundstep).

**Comment by covertg** [comment]
FWIW I suppose that this behavior exists in order to make sure that tuning gets "restarted" once one chain has finished, since single process sampling uses the same step object chain after chain. So order to make this behave the same as multiprocessing sampling, somewhere the original value(s) of `tune` passed to each step method would need to be cached, then accessed whenever a new chain begins.

**Comment by covertg** [comment]
Ah, somehow this didn't turn up in my original search for issues: it looks like a similar issue was present in pymc3. #4886 sought to address it but the draft PR was never quite finished.

It looks like the proposed change in that PR was to create a `SettingsReseter` class that every step method creates an instance of, and subsequently uses to restore step method state when a new chain is begun in serial.
